File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    element = ['A', 'E', 'I', 'O', 'U']
    maxlen = 5
    
    all_words = []
    
    for i in range(1, maxlen + 1):
        for j in product(element, repeat=i):
            all_words.append("".join(j))
            
    all_words.sort()

    return all_words.index(word) + 1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<char> elements = {'A', 'E', 'I', 'O', 'U'};
vector<string> all_words;

void dfs(string current, int max_len) {
    if (!current.empty()) {
        all_words.push_back(current);
    }
    
    if (current.length() == max_len) {
        return;
    }
    
    for (int i = 0; i < elements.size(); i++) {
        dfs(current + elements[i], max_len);
    }
}

int solution(string word) {
    dfs("", 5);

    sort(all_words.begin(), all_words.end());
    
    int index = find(all_words.begin(), all_words.end(), word) - all_words.begin();
    
    return index + 1;
}